Owncloud iOs Client/Login/Login/UniversalLoginViewController.swift

```diff
@@ -933,24 +922,31 @@ public enum TextfieldType: String {
                     
                 } else {
 
+                    //1.Generate correct user
+                    if self.loginMode == .create {
+                        self.user = UserDto()
+                    }
+                    
                     if self.loginMode == .create || self.loginMode == .migrate || self.forceAccountMigration {
-                        let newUser = UserDto()
+
                         if self.loginMode != .create {
-                            newUser.userId = self.user!.userId
-                            if self.loginMode == .migrate {
-                                newUser.predefinedUrl = k_default_url_server
-                            }
+                            self.userNewCredentials.userId = String(self.user!.userId)
                         }
-                        newUser.url = self.validatedServerURL
-                        newUser.username = self.userNewCredentials.userName
-                        newUser.ssl = self.validatedServerURL.hasPrefix("https")
-                        newUser.urlRedirected = app.urlServerRedirected
-                        
-                        self.user = newUser.copy() as? UserDto
+                        self.user!.url = self.validatedServerURL
```

Comment:
We need the user to be a global variable?
Also i think maybe you could extract this into a function to create a more readable and more testable code.